### PR TITLE
compress copy simd

### PR DESCRIPTION
**Pull request by DenisYaroshevskiy**
This should, in theory, finish the implementation of compress_copy itself:
both scalar and simd versions are done and working.

Next patch would be to update the usage.

### CODE REVIEW

File: test/unit/api/compress/compress_store.cpp [hunk 1]
```diff
@@ -0,0 +1,11 @@
+//==================================================================================================
+/**
+  EVE - Expressive Vector Engine
+  Copyright : EVE Project Contributors
+  SPDX-License-Identifier: BSL-1.0
+**/
+//==================================================================================================
+
+#include "test.hpp"
+
```
Comment (jfalcou) [inline test/unit/api/compress/compress_store.cpp:10]: why the empty file then ?

Comment (DenisYaroshevskiy) [reply test/unit/api/compress/compress_store.cpp:10]: I figured so that when one comes looking, the find this

Comment (jfalcou) [reply test/unit/api/compress/compress_store.cpp:10]: well it's an empty target that just consume test compilation time no ?

**Review by jfalcou — APPROVED** [review]

File: test/unit/api/compress/compress_store.cpp (hunk 1 above)
Comment (DenisYaroshevskiy) [reply test/unit/api/compress/compress_store.cpp:10]: Agreed that I'll delete it in the next patch

**Pull request merged**